### PR TITLE
Add support for cracking DHCPv4 authentication

### PR DESCRIPTION
**Pull request by kholia**
This is untested as we don't have a suitable .pcap file. Also, it is not trivial to generate a suitable .pcap file!

This is related to https://github.com/magnumripper/JohnTheRipper/issues/3275 (Add support for cracking DHCP authentication).

### CODE REVIEW

File: run/network2john.lua [hunk 1]
```diff
@@ -257,6 +257,45 @@ end
...
+
+	local algorithm = f_algorithm()
+	if algorithm.value ~= 1 then
+		return
+	end
+
+	local hash = f_hash()
+	if hash == nil then
+		return
+	end
```
[the remark this answers is not in the document]
Comment (kholia) [reply run/network2john.lua]: Good catch. This should be fixed now.

[the remark this answers is not in the document]
Comment (magnumripper) [reply run/network2john.lua]: Same issue with ipv6 if you'd like to clean it up. Merge at will.

[the remark this answers is not in the document]
Comment (magnumripper) [reply run/network2john.lua]: I seem to recall lua is a grandchild of rexx? If so, I've got 30+ years of "experience" 😂🤣

[the remark this answers is not in the document]
Comment (kholia) [reply run/network2john.lua]: Ack. Fixed now as well.

**Pull request merged**